File: domain/src/main/java/com/agendify/domain/records/Estabelecimento.java
package com.agendify.domain.records;

import com.agendify.domain.entities.Servico;

import java.util.List;
import java.util.UUID;

public record Estabelecimento(
        UUID id,
        String nome,
        String urlFotoPerfil,
        String email,
        String senha,
        Long tipo,
        Endereco endereco,
        String cnpj,
        String descricao,

        List<ServicoResponse> servicos,
        List<PeriodoAtendimentoDTO> periodosAtendimento
) {
}

File: frontend/src/app/model/response/Estabelecimento.ts
import {Endereco} from "./Endereco";
import {Servico} from "./Servico";
import {PeriodoAtendimento} from "./PeriodoAtendimento";

export interface Estabelecimento {
  id: string;
  nome: string;
  urlFotoPerfil: string;
  senha:string;
  email: string;
  tipo: number;
  endereco: Endereco;
  cnpj: string;
  descricao: string;
  servicos: Servico[];
  periodosAtendimento:PeriodoAtendimento[];
}
